micro-rdk/src/common/ota.rs

```diff
@@ -334,14 +327,32 @@ impl<S: OtaMetadataStorage> OtaService<S> {
             uri = hyper::Uri::from_parts(parts).map_err(|e| OtaError::Other(e.to_string()))?;
         };
 
+        Ok(uri)
+    }
+
+    /// Attempts to perform an OTA update.
+    /// On success, returns an `Ok(true)` or `Ok(false)` indicating if a reboot is necessary.
+    pub(crate) async fn update(&mut self) -> Result<bool, OtaError<S>> {
+        if !(self.needs_update()) {
+            return Ok(false);
+        }
+
+        let mut uri = self.parse_uri(&self.url)?;
+        let mut response: hyper::Response<hyper::body::Incoming>;
+        let mut conn;
+
         let mut num_tries = 0;
-        let (mut sender, conn) = loop {
+        loop {
             num_tries += 1;
             if num_tries == NUM_RETRY_CONN + 1 {
                 return Err(OtaError::Other(
                     "failed to establish connection".to_string(),
                 ));
             }
+
+            let mut sender = None;
```

Comment:
[optional] - Might not be a bad idea to `log::info` here that we are on the N attempt to do an OTA download to the current URL.